testbeam_analysis/hit_analysis.py

```diff
@@ -63,22 +67,76 @@ def remove_noisy_pixels(input_hits_file, n_pixel, output_hits_file=None,
     blurred = median_filter(occupancy.astype(np.int32), size=filter_size, mode='constant', cval=0.0)
     # Spot noisy pixels maxima by substracting expected occupancy
     difference = np.ma.masked_array(occupancy - blurred)
-
     std = np.ma.std(difference)
     abs_occ_threshold = threshold * std
     occupancy = np.ma.masked_where(difference > abs_occ_threshold, occupancy)
     logging.info('Removed %d hot pixels at threshold %.1f in %s', np.ma.count_masked(occupancy), threshold, input_hits_file)
-
     # Generate tuple col / row array of hot pixels, do not use getmask()
     noisy_pixels_mask = np.ma.getmaskarray(occupancy)
+
+    with tb.open_file(output_mask_file, 'w') as out_file_h5:
+        # Creating occupancy table without masking noisy pixels
+        occupancy_array_table = out_file_h5.create_carray(out_file_h5.root, name='HistOcc', title='Occupancy Histogram', atom=tb.Atom.from_dtype(occupancy.dtype), shape=occupancy.shape, filters=tb.Filters(complib='blosc', complevel=5, fletcher32=False))
```

Comment:
Maybe it is more convenient to add the noisy / deactivated pixel information as a user attribute to the Hits node? Then we do not need to deal with an additional file and additional function parameters. Python objects can be easily pickled by pytables and are even visible in vitables.